samza-core/src/main/java/org/apache/samza/container/grouper/task/TaskNameGrouper.java

```diff
@@ -44,15 +43,36 @@
  * </p>
  */
 public interface TaskNameGrouper {
+
+  /**
+   * Groups the taskModels into set of {@link ContainerModel} using the provided grouperContext.
+   *
+   * @param taskModels the set of tasks to group into containers.
+   * @param grouperContext the existing historical metadata of the samza job.
+   * @return the grouped {@link ContainerModel} built from the provided taskModels.
+   */
+  default Set<ContainerModel> group(Set<TaskModel> taskModels, GrouperContext grouperContext) {
+    return group(taskModels);
+  }
+
   /**
-   * Group tasks into the containers they will share.
+   * Group the taskModels into set of {@link ContainerModel}.
    *
-   * @param tasks Set of tasks to group into containers.
-   * @return Set of containers, which contain the tasks that were passed in.
+   * @param taskModels the set of {@link TaskModel} to group into containers.
+   * @return the grouped {@link ContainerModel} built from the provided taskModels.
    */
-  Set<ContainerModel> group(Set<TaskModel> tasks);
+  @Deprecated
+  Set<ContainerModel> group(Set<TaskModel> taskModels);
```

Comment:
Why are we deprecating these? Are we expecting that non-host affinity jobs also require some metadata from the grouper context to perform grouping?